src/JukeBox.cpp

```diff
@@ -67,18 +68,23 @@ TResult<vector<BaseTrack>> JukeBox::queryTracks(string const &searchPattern,
 
 TResult<QueueStatus> JukeBox::getCurrentQueues(TSessionID const &) {
   QueueStatus qs;
-  //  qs.adminQueue = mDataStore.getQueue(QueueType::Admin);
-  //  if (holds_alternative<Error>(qs.adminQueue))
-  //    return get<Error>(qs.adminQueue);
-  //
-  //  qs.normalQueue = mDataStore.getQueue(QueueType::Normal);
-  //  if (holds_alternative<Error>(qs.normalQueue))
-  //    return get<Error>(qs.normalQueue);
-  //
-  //  qs.currentTrack = mDataStore.getPlayingTrack();
-  //  if (holds_alternative<Error>(qs.currentTrack))
-  //    return get<Error>(qs.currentTrack);
-  //
+
+  auto ret = mDataStore->getQueue(QueueType::Admin);
+  if (holds_alternative<Error>(ret))
+    return get<Error>(ret);
+  qs.adminQueue = get<Queue>(ret);
+
+  ret = mDataStore->getQueue(QueueType::Normal);
+  if (holds_alternative<Error>(ret))
+    return get<Error>(ret);
+  qs.normalQueue = get<Queue>(ret);
+
+  auto track = mDataStore->getPlayingTrack();
+  if (holds_alternative<Error>(track))
+    return get<Error>(track);
+  // TODO: How can i assign a QueuedTrack on a PlaybackTrack?
```

Comment:
would not be a problem if `getPlayingTrack` actually returned a playback track..

Reply:
maybe query the music backend for the currently playing song and merge that with the track from the datastore if necessary